apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -567,12 +601,33 @@ def grade(element_html: str, data: pl.QuestionData) -> None:
                 depends_graph[tag] = cur_rank_depends
                 prev_rank = ranking
 
-        elif grading_method is GradingMethodType.DAG:
+            num_initial_correct, true_answer_length = grade_dag(
+                submission, depends_graph, group_belonging
+            )
+        elif (
+            grading_method is GradingMethodType.DAG
+            and not order_blocks_options.is_optional
+        ):
             depends_graph, group_belonging = extract_dag(true_answer_list)
+            num_initial_correct, true_answer_length = grade_dag(
+                submission, depends_graph, group_belonging
+            )
+        elif (
+            grading_method is GradingMethodType.DAG and order_blocks_options.is_optional
+        ):
+            # extract multigraph from all blocks not just those in correct_answer
+            depends_multigraph, final = extract_multigraph(true_answer_list)
+
+            # TODO: add group belonging support for group blocks
```

Comment:
I don't think we want this as a TODO in the code, its listed on an issue here anyway: #13124